File: IT/Multithreading/Esercizi/Java/Esercizio1.java
/* file Esercizio1.java */

class ChildThread extends Thread {
	@Override
	public void run() {
		try {
			String threadName = getName();
			System.out.println("Inizio del thread " + threadName + ".");
			sleep(5000);
			System.out.println("Fine del thread " + threadName + ".");
		}
		catch (InterruptedException e) {
			e.printStackTrace();
		}
	}
}

public class Esercizio1 {
	public static void main(String [] args) {
		try
		{
			String main = Thread.currentThread().getName();
			System.out.println("Thread " + main);

			Thread t = new ChildThread();
			t.start();
			t.join();

			System.out.println("Fine del thread " + main + ".");
		}
		catch (InterruptedException e) {
			e.printStackTrace();
		}
	}
}

File: IT/Multithreading/Esercizi/Python/esercizio1.py
# file esercizio1.py

from __future__ import print_function
from multiprocessing import Process, current_process
from time import sleep

def func():
	pName = current_process().name
	print("Inizio del processo {}.".format(pName))
	sleep(5)
	print("Fine del processo {}.".format(pName))

if __name__ == "__main__":
	main = current_process().name
	print("Processo {}".format(main))

	p = Process(target=func)

	p.start()
	p.join()

	print("Fine del processo {}.".format(main))

